martin-mbtiles/src/mbtiles.rs

```diff
@@ -268,6 +274,31 @@ impl Mbtiles {
         }
         Ok(None)
     }
+
+    pub async fn detect_type<T>(&self, conn: &mut T) -> MbtResult<Type>
+    where
+        for<'e> &'e mut T: SqliteExecutor<'e>,
+    {
+        let is_tiles_tables_type: i32 =
+            query(include_str!("queries/is_tile_tables_mbtiles_type.sql"))
```

Comment:
lets make the filenames just a bit shorter - they are all under mbtiles dir, so perhaps `sql/is_tile_tables_type.sql` ?